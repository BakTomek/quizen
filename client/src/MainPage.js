import React from 'react';

const MainPage = () => {
  return (
    <div>
      <h1>Welcome to the Quiz App</h1>
      <p>Select a tab from the menu to get started.</p>
    </div>
  );
};

export default MainPage;
