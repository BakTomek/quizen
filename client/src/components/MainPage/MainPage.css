#MainPageDiv{
    width: 100%;
    height: 99%;
    padding: 15px;
    overflow-y: scroll;
    background-color: #E9EFF2;
    display: flex;
    justify-content: safe center;
    align-items: safe center;
    flex-direction: column;
}

#MainPageDiv h1{
    color: #1B4863;
    font-size: 40px;
    line-height: 48px;
    letter-spacing: -1%;
    margin-bottom: 72px;
}

#Arrow{
    max-width: 200px;
    max-height: 200px;
}

#ArrowDiv{
    width: 90%;
    max-height: 1500px;
    color: #E6CC65;
    font-size: 26px;
    line-height: 32px;
    letter-spacing: -1%;
    font-weight: 700;
    display: flex;
    justify-content: left;
    align-items: left;
    flex-direction: row;
    flex-wrap: wrap;
}

#ArrowDiv p{
    padding-left: 100px;
}

.fill{
    width: 450px;
    height: 300px;
    display: flex;
    justify-content: right;
    align-items: center;
    color: #02090D;
}

#hat{
    max-width: 32px;
    max-height: 32px;
    margin-right: 10px;
}

#BottomFill{
    height: 500px;
    width: 80%;
}