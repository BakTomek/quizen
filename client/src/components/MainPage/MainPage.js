import React from 'react';
import './MainPage.css';
import arrow from './images/arrow.png'
import back from './background_bruh_2.png'

const divStyle = {
  backgroundImage: `url(${back})`,
  backgroundSize: 'cover',
  backgroundRepeat: 'no-repeat',
  backgroundPosition: 'center'
};

const MainPage = () => {
  return (
    <div id='MainPageDiv' data-testid='main-page-div' style={divStyle}>
      <h1>Welcome to the Quiz App</h1>
      <div id='ArrowDiv' data-testid='arrow-div'>
        <div>
          <img id='Arrow' data-testid='arrow-image' alt='Arrow' src={arrow}/>
          <p>Select a tab from the menu to get started. </p>
        </div>
        <div className='fill'></div>
      </div>
      <div id='BottomFill' data-testid='bottom-fill'></div>
    </div>
  );
};

export default MainPage;
