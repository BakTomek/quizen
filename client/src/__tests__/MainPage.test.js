import React from 'react';
import { render, screen } from '@testing-library/react';
import MainPage from '../components/MainPage/MainPage.js';

describe('MainPage Component', () => {
  test('it match snapshot', () => {
    const {toJSON} = render(<MainPage/>)
    expect(toJSON()).toMatchSnapshot();
  })
  test('renders MainPage correctly', () => {
    const {getByTestId} = render(<MainPage />);

    expect(screen.getByText('Welcome to the Quiz App')).toBeInTheDocument();
    expect(screen.getByText('Select a tab from the menu to get started.')).toBeInTheDocument();

    const arrowImage = document.getElementById('Arrow');
    expect(arrowImage).toBeInTheDocument();

    const mainPageDiv = getByTestId('main-page-div');
    expect(mainPageDiv).toBeInTheDocument();
    // expect(mainPageDiv).toHaveStyle(`backgroundImage: url(mockBackground.png)`);
    // expect(mainPageDiv).toHaveStyle('backgroundSize: cover');
    // expect(mainPageDiv).toHaveStyle('backgroundRepeat: no-repeat');
    // expect(mainPageDiv).toHaveStyle('backgroundPosition: center');

    expect(screen.getByTestId('arrow-div')).toBeInTheDocument();
    expect(screen.getByTestId('bottom-fill')).toBeInTheDocument();
  });
});
