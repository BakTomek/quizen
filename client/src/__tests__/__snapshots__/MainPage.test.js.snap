// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="MainPageDiv"
    style="background-image: url(); background-size: cover; background-repeat: no-repeat; background-position: center;"
  >
    <h1>
      Welcome to the Quiz App
    </h1>
    <div
      id="ArrowDiv"
    >
      <div>
        <img
          alt="Arrow"
          id="Arrow"
          src=""
        />
        <p>
          Select a tab from the menu to get started. 
        </p>
      </div>
      <div
        class="fill"
      />
    </div>
    <div
      id="BottomFill"
    />
  </div>
</DocumentFragment>
`;
